### DISCUSSION TITLE
Fetching model lists from OpenAI failed when http forward proxy is set

### DISCUSSION THREAD

**Discussion by stargazerZJ** [danny-avila/LibreChat#4690] [Troubleshooting]
### What happened?

`.env` config:
```bash
OPENAI_REVERSE_PROXY=https://api.my-oai-reverse-proxy.com/v1
```

`error-2024-xx-xx.log`:
```
{"data":"Client sent an HTTP request to an HTTPS server.\n","headers":{"connection":"close"},"level":"error","message":"Failed to fetch models from https://api.my-oai-reverse-proxy.com API The request was made and the server responded with a status code that falls out of the range of 2xx: Request failed with status code 400. Error response data:\n","status":400}
```

This works
```bash
$ curl https://api.my-oai-reverse-proxy.com/v1/models \ 
  -H "Authorization: Bearer $OPENAI_API_KEY"
  {
  # json data
  }
  ```
Using http will lead to error, but strangely, not `400`:
  ```bash
  curl http://api.my-oai-reverse-proxy.com/v1/models -v 
  # some info omitted
< HTTP/1.1 308 Permanent Redirect
< Location: https://api.my-oai-reverse-proxy.com/v1/models
< Content-Length: 0
<
* Closing connection 0
```

Thank you for your excellent project!

### Steps to Reproduce

1. use docker compose to set up
2. update to the latest version
3. configure the openai reverse proxy

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
This has more to do with your proxy than the project

**Comment by stargazerZJ** [comment]
Upon further testing, I find that the problem has no relationship with the reverse openai proxy, but is caused by the forward proxy I set in the `.env` file:
```
PROXY=http://172.17.0.1:1080
```

It appears that when fetching models, the request sent to the http forward proxy is somewhat malformed, causing the http proxy to return a `400 Bad Request`.

I confirmed that this problem exist when only `PROXY` is set, and `OPENAI_REVERSE_PROXY` is not set.

I'm happy to investigate this case further.
